tools/distpartitioning/data_shuffle.py

```diff
@@ -520,17 +524,21 @@ def gen_dist_partitions(rank, world_size, params):
     #init processing
     schema_map = read_json(os.path.join(params.input_dir, params.schema))
 
-    #TODO: For large graphs, this mapping function can be memory intensive. This needs to be changed to 
-    #processes owning a set of global-nids, per partitioning algorithm, and messaging will be used to 
-    #identify the ownership instead of mem. lookups. 
-    node_part_ids = read_ntype_partition_files(schema_map, os.path.join(params.input_dir, params.partitions_dir))
+    #Initialize distributed lookup service for partition-id and shuffle-global-nids mappings
+    #for global-nids
+    _, global_nid_ranges = get_idranges(schema_map[constants.STR_NODE_TYPE], 
+                                        schema_map[constants.STR_NUM_NODES_PER_CHUNK])
+    id_map = dgl.distributed.id_map.IdMap(global_nid_ranges)
+    id_lookup = DistLookupService(os.path.join(params.input_dir, params.partitions_dir),\
+                                    schema_map[constants.STR_NODE_TYPE],\
+                                    id_map, rank, world_size)
 
     ntypes_ntypeid_map, ntypes, ntypeid_ntypes_map = get_node_types(schema_map)
     print('[Rank: ', rank, '] Initialized metis partitions and node_types map...')
 
     #read input graph files and augment these datastructures with
     #appropriate information (global_nid and owner process) for node and edge data
-    node_tids, node_features, node_feat_tids, edge_data, edge_features = read_dataset(rank, world_size, node_part_ids, params, schema_map)
+    node_tids, node_features, node_feat_tids, edge_data, edge_features = read_dataset(rank, world_size, id_lookup, params, schema_map)
```

Comment:
Break this super long line? The suggested max length is 79 / 80
https://peps.python.org/pep-0008/#maximum-line-length
https://google.github.io/styleguide/pyguide.html#s3.2-line-length

Reply:
Done. 

Da suggested to run lint and make all the suggested changes by lint in a separate PR -- so that it will be easy to do code review. 

Will run lint and open a separate PR to address these lint issues.